### ISSUE TITLE
Disabled checkboxes should not be tab navigable

### ISSUE THREAD

**Issue by zepumph** [phetsims/sun#517]
From https://github.com/phetsims/inverse-square-law-common/issues/66#issuecomment-509274593

It seems like though the checkbox is properly disabled and can't be toggled, when a checkbox is not enabled, you can still tab navigate to it, and it has focus highlight.

See here: http://localhost/sun/sun_en.html?brand=phet&ea&a11y&screens=2&component=Checkbox

![image](https://user-images.githubusercontent.com/6856943/60829839-5bc4c580-a162-11e9-998e-b11aada5cbf0.png)

@jessegreenberg in html, how do you think the best way to accomplish this is? From a small bit of poking, it seems like toggling the "disabled" attribute based on the state of the enabledProperty will probably work out nicely.

Also note that this is not solely a checkbox problem, but any PhET component that runs on enabledProperty. Likely this is code that we should not spend too much time implementing until it can be added in a single place: to the solution in https://github.com/phetsims/sun/issues/257 rather than scattered all over common code.

**Comment by zepumph** [comment]
Here is a patch that I believe does what we are interested in accomplishing. I didn't commit because it wasn't totally trivial, and I would rather just use it as a discussion point to try to boost the priority of https://github.com/phetsims/sun/issues/257

```diff
Index: js/Checkbox.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/Checkbox.js	(revision e0a45b5ebad14e37eec8c53fbbc306a7e8933a15)
+++ js/Checkbox.js	(date 1562607825480)
@@ -160,7 +160,13 @@
     } );
 
     var enabledListener = function( enabled ) {
-      !enabled && self.interruptSubtreeInput(); // interrupt interaction
+      if ( enabled ) {
+        self.hasAccessibleAttribute( 'disabled' ) && self.removeAccessibleAttribute( 'disabled' );
+      }
+      else {
+        self.interruptSubtreeInput(); // interrupt interaction
+        self.setAccessibleAttribute( 'disabled', true );
+      }
       self.pickable = enabled;
       self.opacity = enabled ? 1 : options.disabledOpacity;
     };
```

**Comment by zepumph** [comment]
See https://github.com/phetsims/gravity-force-lab/issues/110 for the GFLB main issue.

**Comment by jessegreenberg** [comment]
Proposed change looks very good, however I recall that last we discussed this we decided that disabled components should be focusable but not interactive. I don't remember why we decided that because that is different from the native `disabled` attribute, but I am going to try to find any discussion notes about it.

**Comment by jessegreenberg** [comment]
Found it! https://github.com/phetsims/sun/issues/317 (specifically https://github.com/phetsims/sun/issues/317#issuecomment-333263613)

I think all we need to do is replace `disabled` in the proposed change with `aria-disabled`.

**Comment by zepumph** [comment]
Gotcha! Thanks for looking into this.

**Comment by zepumph** [comment]
We don't need this done until GFL is published, so I think for now this should be put on hold until https://github.com/phetsims/sun/issues/257 is done. We can take it off of hold before then if we are getting close to publishing that sim.

**Comment by zepumph** [comment]
Actually I think we should mark this off hold. Especially with https://github.com/phetsims/sun/issues/519, we should just solve these problems now, together as a batch.

**Comment by zepumph** [comment]
Over in https://github.com/phetsims/sun/issues/519 we found a solution that can keep diabled checkboxes in the tab order. Based on previous discussion referenced in https://github.com/phetsims/sun/issues/517#issuecomment-509331910 it seems like that is the way to go. Closing
